### ISSUE TITLE
determine after or before in "usage" instead of the declaration --edit

### ISSUE THREAD

**Issue by ciroreed** [ciroreed/kaop#1]

**Comment by ciroreed** [comment]
in fact, its better to define "hooks".. i need to keep this. What im looking is just have the possibility to be more expressive. If we can optionally declare the annotation at the end of the function we're making our code more readable. 

IE: 

```javascript
var service = Class.static({
  actionWhichNeedsToBeLoggedAfterExecution: [function(){}, "$log"],
  actionWhichNeedsToBeLoggedBeforeExecution: ["$log", function(){}]
})
```

**Comment by ciroreed** [comment]
this issue was solved by 04b82f59a4c3a4adeef826de4742b02720750927
